server/sonar-server/src/test/java/org/sonar/server/projectlink/ws/CreateActionTest.java

```diff
@@ -0,0 +1,188 @@
+  private final String PROJECT_UUID = UUID_EXAMPLE_01;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+  DbClient dbClient = db.getDbClient();
+  DbSession dbSession = db.getSession();
+
+  WsActionTester ws;
+
+  CreateAction underTest;
+
+  @Before
+  public void setUp() {
+    ComponentFinder componentFinder = new ComponentFinder(dbClient);
+    underTest = new CreateAction(dbClient, userSession, componentFinder);
+    ws = new WsActionTester(underTest);
+
+    userSession.login("login").setGlobalPermissions(SYSTEM_ADMIN);
+  }
+
+  @Test
+  public void test_example_with_key() {
+    insertProject();
+
+    String result = ws.newRequest()
+      .setMethod("POST")
+      .setParam(PARAM_PROJECT_KEY, PROJECT_KEY)
+      .setParam(PARAM_NAME, "Custom")
+      .setParam(PARAM_URL, "http://example.org")
+      .execute().getInput();
+
+    assertJson(result).ignoreFields("id").isSimilarTo(getClass().getResource("create-example.json"));
+  }
+
+  @Test
+  public void test_example_with_id() {
+    insertProject();
+
+    String result = ws.newRequest()
+      .setMethod("POST")
+      .setParam(PARAM_PROJECT_ID, PROJECT_UUID)
+      .setParam(PARAM_NAME, "Custom")
+      .setParam(PARAM_URL, "http://example.org")
+      .execute().getInput();
+
+    assertJson(result).ignoreFields("id").isSimilarTo(getClass().getResource("create-example.json"));
+  }
+
+  @Test
+  public void test_if_global_admin() throws IOException {
+    userSession.login("login").setGlobalPermissions(SYSTEM_ADMIN);
+    insertProject();
+    createAndTest();
+  }
+
+  @Test
+  public void test_if_project_admin() throws IOException {
+    userSession.login("login");
+    ComponentDto project = insertProject();
+    userSession.addProjectUuidPermissions(UserRole.ADMIN, project.uuid());
+    createAndTest();
+  }
+
+  @Test
+  public void fail_when_no_project() {
+    expectedException.expect(NotFoundException.class);
+    ws.newRequest()
+      .setParam(PARAM_PROJECT_KEY, "unknown")
+      .setParam(PARAM_NAME, "Custom")
+      .setParam(PARAM_URL, "http://example.org")
+      .execute();
+  }
+
+  @Test
+  public void fail_if_anonymous() {
+    userSession.anonymous();
+    insertProject();
+
+    expectedException.expect(ForbiddenException.class);
+    ws.newRequest()
+      .setParam(PARAM_PROJECT_KEY, PROJECT_KEY)
+      .setParam(PARAM_NAME, "Custom")
+      .setParam(PARAM_URL, "http://example.org")
+      .execute();
+  }
+
+  @Test
+  public void fail_if_not_project_admin() {
+    userSession.login("login");
+    insertProject();
+
+    expectedException.expect(ForbiddenException.class);
+    ws.newRequest()
+      .setParam(PARAM_PROJECT_KEY, PROJECT_KEY)
+      .setParam(PARAM_NAME, "Custom")
+      .setParam(PARAM_URL, "http://example.org")
+      .execute();
+  }
+
```

Comment:
please add a test that checks it fails when:

-  name is not provided
- url is not provided